sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -1327,22 +1327,29 @@ class AstBuilder(conf: SQLConf) extends SqlBaseBaseVisitor[AnyRef] with Logging
   override def visitComparison(ctx: ComparisonContext): Expression = withOrigin(ctx) {
     val left = expression(ctx.left)
     val right = expression(ctx.right)
+    val isUnsupportedComparison = (left, right) match {
```

Comment:
Total nit, but isn't it clearer to write `isSupportedComparison` and avoid inverting the logic everywhere?